### ISSUE TITLE
Recent mzcompose refactoring breaks the handling of `:latest` images

### ISSUE THREAD

**Issue by philip-stoev** [MaterializeInc/materialize#10713]
### What version of Materialize are you using?

v0.20.0

### How did you install Materialize?

Built from source

### What was the issue?

Running the feature benchmark with `--this-tag=latest` no longer causes `git pull` to be performed to fetch the latest `materialized:latest` container. An older version of `latest` will be used if available locally. This may cause the feature benchmark to randomly report performance issues where none exist.

This was previously fixed in:

```
commit fb2aef8224133848d7b1263c1fe3a7c105c01676
Merge: c0db247c7 e6689f4e9
Author: Philip Stoev <pstoev@materialize.io>
Date:   Wed Jan 26 18:34:38 2022 +0200

    Merge pull request #10251 from philip-stoev/mzcompose-pull-latest

    mzbuild: always perform docker pull on acquire()
```

but now seems broken again due to some recent refactoring. There are at least two problems I could see:

1. `mzbuild.py:acquire` does not have a special case for handling `latest` so it may return early if it considers this image to be available already, regardless of whether it is truly the latest version or not

2. `_munge_services` plus the folow-up `acquire()` in `overrride()` will not cause actual image acquisition. I tried to debug this and it seems that the entire resolving procedure considers the `materialized:latest` to not be a dependency to anything so it will not be docker-pull-ed, as only dependencies are fetched, rather than all participating images.

I tried to fix this but I could not arrive at a satisfactory patch , so @benesch your assistance would be much appreciated.

**Comment by benesch** [comment]
What about something like this, @philip-stoev?

```diff
diff --git a/misc/python/materialize/mzcompose/__init__.py b/misc/python/materialize/mzcompose/__init__.py
index 235e681c1..2a0b5b9b4 100644
--- a/misc/python/materialize/mzcompose/__init__.py
+++ b/misc/python/materialize/mzcompose/__init__.py
@@ -568,6 +568,14 @@ class Composition:
             capture=capture,
         )
 
+    def pull(self, *services: str) -> None:
+        """Pull service images.
+
+        Args:
+            services: The names of services in the composition.
+        """
+        self.invoke("pull", *services)
+
     def up(self, *services: str, detach: bool = True) -> None:
         """Build, (re)create, and start the named services.
 
diff --git a/test/feature-benchmark/mzcompose.py b/test/feature-benchmark/mzcompose.py
index 3120fb9f6..73d887ae7 100644
--- a/test/feature-benchmark/mzcompose.py
+++ b/test/feature-benchmark/mzcompose.py
@@ -116,6 +116,9 @@ def run_one_scenario(
 
     for mz_id, instance in enumerate(["this", "other"]):
         with c.override(mzs[instance]):
+            if mzs[instance].image:
+                c.pull("materialized")
+
             print(f"The version of the '{instance.upper()}' Mz instance is:")
             c.run("materialized", "--version")
 

```

**Comment by benesch** [comment]
It won't apply to things besides the feature benchmark, but it is a very small and targeted patch!

**Comment by philip-stoev** [comment]
While I agree that it is a small patch, it breaks the layering by having the test be responsible for image management. Please consider my proposal at https://github.com/MaterializeInc/materialize/pull/10808/

**Comment by benesch** [comment]
Fixed by #10808.
